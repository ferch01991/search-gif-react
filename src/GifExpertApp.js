import React, { useState } from 'react'

import { AddCategory } from "./AddCategory";
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball'])

    const handlerAdd = (e) => {
        // Spread extrae todos los elementos
        // setCategories([...categories, 'Hunter'])
        // dentro del setCategories recibe tambien un callback
        setCategories(cats => [...cats, 'Hunter'])
        
    }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
            {/* <button onClick={handlerAdd}>Agregar</button> */}

            <ol>
                {
                    // El key no tiene qeu ser el indice y debe ser unico
                    categories.map(category => 
                        <GifGrid 
                            key={category}
                            category= { category }
                        />)
                }
            </ol>
        </>
    )
}
