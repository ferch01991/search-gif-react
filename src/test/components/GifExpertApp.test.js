import React from 'react'
import { shallow } from 'enzyme'

import { GifExpertApp } from "./../../GifExpertApp";

describe('Prueba de GifExpertApp', () => {
    
    test('should matrch with snapshop', () => {
        
        const wrapper = shallow( <GifExpertApp /> )
        expect(wrapper).toMatchSnapshot();
    });

    test('should show a category list', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultcatgories={categories} /> )

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
    
    
})
