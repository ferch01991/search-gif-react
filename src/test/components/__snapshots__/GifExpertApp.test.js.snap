// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifExpertApp should matrch with snapshop 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Prueba de GifExpertApp should show a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
